core/src/main/java/org/mule/runtime/core/internal/metadata/DefaultMetadataResolverFactory.java

```diff
@@ -37,20 +38,20 @@
   private final Map<String, InputTypeResolver> inputResolvers = new HashMap<>();
   private final TypeKeysResolver keysResolver;
 
-  public DefaultMetadataResolverFactory(Class<? extends TypeKeysResolver> keyResolver,
-                                        Map<String, Class<? extends InputTypeResolver>> typeResolvers,
-                                        Class<? extends OutputTypeResolver> outputResolver,
-                                        Class<? extends AttributesTypeResolver> attributesResolver) {
+  public DefaultMetadataResolverFactory(Supplier<? extends TypeKeysResolver> keyResolver,
+                                        Map<String, Supplier<? extends InputTypeResolver>> typeResolvers,
+                                        Supplier<? extends OutputTypeResolver> outputResolver,
+                                        Supplier<? extends AttributesTypeResolver> attributesResolver) {
 
     checkArgument(keyResolver != null, "MetadataKeyResolver type cannot be null");
```

Comment:
this doesn't make much sense now... or at least should be validated also for the result of the supplier